modules/logging/filters.go

```diff
@@ -44,20 +52,63 @@ func (DeleteFilter) CaddyModule() caddy.ModuleInfo {
 	}
 }
 
+// UnmarshalCaddyfile sets up the module from Caddyfile tokens.
+func (m *DeleteFilter) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	for d.Next() {
+		for d.NextBlock(0) {
+			switch d.Val() {
+			case "except":
+				m.ExceptRaw = d.RemainingArgs()
+				if len(m.ExceptRaw) < 1 {
+					return d.ArgErr()
+				}
+
+			default:
+				return d.Errf("unrecognized subdirective %s", d.Val())
+			}
+		}
+	}
+	return nil
+}
+
+// Provision converts the except list to a map for fast lookup.
+func (d *DeleteFilter) Provision(ctx caddy.Context) error {
+	d.except = make(map[string]bool)
+	for _, val := range d.ExceptRaw {
+		d.except[val] = true
+	}
+	return nil
+}
+
 // Filter filters the input field.
-func (DeleteFilter) Filter(in zapcore.Field) zapcore.Field {
-	in.Type = zapcore.SkipType
+func (d DeleteFilter) Filter(in zapcore.Field) zapcore.Field {
+	if len(d.except) == 0 {
+		in.Type = zapcore.SkipType
+		return in
+	}
+
+	v := reflect.ValueOf(in.Interface)
+	if v.Kind() == reflect.Map {
```

Comment:
What if it's a struct/object with fields instead of a map with keys?

Reply:
I don't know. I didn't look that deeply to see how that would work. I was primarily concerned with the `header` usecase because that sounds like the most common usecase. Can you think of an example field where that would apply, that I could test for?

Reply:
The `request` field (parent of `header`, I think) is a struct/object. Granted, they're not _as dynamic_ since they're strongly typed, but I can see users might want to delete the whole object except for one property.

Honestly this is fine, we can do more based on feature requests if necessary.

Is this still a draft?

Reply:
If you're happy with this, then cool.

I'm still a bit concerned about the part where we're straight up modifying the `http.Request`, seems "destructive". But if this is always the last thing in the middleware chain, then I guess it's fine? But 😬 

Anyways - we should merge the other `filter` PR first, then I'll un-draft this and rebase, to keep those separate.

Reply:
Okay... thanks for bringing that up again. Yeah, modifying the underlying map is going to be a problem. You can verify this with:

```
respond {header.User-Agent}
```

(Obviously in conjunction with filtered logs.) And I bet you will not see anything in the response, despite clients setting it.

Reply:
Yeah, confirmed. `respond {header.User-Agent}` returns empty when `except` is used.

I don't know how we could do this otherwise...

Reply:
The reason this isn't working is because it's a hack on the design. The proper way to do this is to have some other Caddy app module (or any other external log aggregator service) ingest the logs and filter them after being encoded. I don't know another way to do it.